docs/runtime-code-typescript-setup.md

```diff
@@ -0,0 +1,146 @@
+# TypeScript Setup
+
+The game server includes a JavaScript Virtual Machine (VM) which can be used to load and run custom logic which is specific to your game project. This is in addition to Lua and Go as supported programming languages to write your server code.
+
+It's useful to implement game code you would not want to run on the client or trust the client to provide unchecked inputs on. You can think of this feature of Nakama as similar to what is sometimes called Lambda or Cloud Functions in other systems. A good use case is if you wanted to grant the user a reward each day that they play the game.
+
+TypeScript is a fantastic superset of the JavaScript language. It allows you to write your code with types which helps to reduce bugs and unexpected runtime behaviour of code. Nakama's support for JavaScript has been built to directly consider the use of TypeScript for your code and is the recommended way to develop your JavaScript code.
+
+!!! Note
+    You can learn more about how to write your JavaScript code in TypeScript in the <a href="https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html" target="_blank">official documentation</a>.
+
+## Prerequisites
+
+You will need to have these tools installed to work with TypeScript for your project:
+
+* Node v14 (active LTS) or greater.
+* Basic UNIX tools or knowledge on the Windows equivalents.
+
+The TypeScript compiler and other dependencies will be fetched with NPM.
+
+## Initialize the Project
+
+These steps will set up a workspace to write all your project code to be run by the game server.
+
+1. Define the folder name that will be the workspace for the project. In this case we'll use "ts-project".
+
+    ``` shell
+    mkdir -p ts-project/{src,build}
+    cd ts-project
+    ```
+
+2. Use NPM to set up the Node dependencies in the project. Install the TypeScript compiler.
+
+    ``` shell
+    npm init -y
+    npm install --save-dev typescript
+    ```
+
+3. Use the TypeScript compiler installed to the project to set up the compiler options.
+
+    ``` shell
+    npx tsc --init
+    ```
+
+    You'll have a "tsconfig.json" file describe the available options that can be added run on the TypeScript compiler. When you've trimmed the commented out entries and updated it a minimal file will look something like:
+
+    ``` json
+    {
+      "compilerOptions": {
```

Comment:
Do we need these to be the specific settings?

Reply:
These are the default settings generated by `tsc init` with the `"module": "commonjs"` option removed because it cannot be used with the `outFile` option. I think we should show this file because it will make the setup clearer.